File: 11_01_2016_two_sum/chang.cpp
class Solution {
public:
    /*
     * @param numbers : An array of Integer
     * @param target : target = numbers[index1] + numbers[index2]
     * @return : [index1+1, index2+1] (index1 < index2)
     */
        vector<int> twoSum(vector<int>& nums, int target) {
        // 
        unordered_map<int, int> hash;
        vector<int> result;

        // 
        for (int i = 0; i < nums.size(); i++) {
            if (hash.find(target - nums[i]) != hash.end()) {
                // 
                result.push_back(hash[target - nums[i]] + 1);
                result.push_back(i + 1);
                return result;
            }
            hash[nums[i]] = i;
        }

        // 
        result.push_back(-1);
        result.push_back(-1);
        return result;
    }
};

File: 11_08_2016_climbing_stairs/chang.py
class Solution:
    """
    @param n: An integer
    @return: An integer
    """
    def climbStairs(self, n):
        if n < 2:
            return 1
            
        fb = [1 for x in range(n+1)]
        
        for i in range(2,n+1):
            fb[i] = fb[i-1] + fb[i-2]
        
        return fb[n]
